boards/arm/nrf5340dk_nrf5340/Kconfig.defconfig

```diff
@@ -14,6 +14,17 @@ config BOARD
 config BUILD_WITH_TFM
 	default y if BOARD_NRF5340DK_NRF5340_CPUAPPNS
 
+if BUILD_WITH_TFM
+
+# By default, if we build with TF-M, instruct build system to
+# flash the combined TF-M (Secure) & Zephyr (Non Secure) image
+# (when building in-tree tests).
+config TFM_FLASH_MERGED_BINARY
```

Comment:
Should this be a general setting ?
As Zephyr already supports the property `HEX_FILES_TO_MERGE` then why not use that instead of:
https://github.com/zephyrproject-rtos/zephyr/blob/master/modules/trusted-firmware-m/CMakeLists.txt#L309-L314

and then make this PR into a generic option in Zephyr to flash the `merged.hex` instead of `zephyr.hex` if present ?